### PR TITLE
Controllers

### PR DESCRIPTION
**Pull request by jsug9**
### Hello 👋 
**In this project:**
-  For each URL I created:
    - A route.
    - An action in the correct controller.
    - A view file.

-  View files should include only placeholders written in plain HTML.

-  Added code to handle the 4 required URLs.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -3,4 +3,11 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+
+  root "users#index"
+
+  get '/users/:author_id/posts' => 'posts#index', as: :user_posts
+  get '/users/:author_id/posts/:id' => 'posts#show', as: :user_post
+  get '/users' => 'users#index'
+  get '/users/:id' => 'users#show'
```
Comment (gemmen29) [inline config/routes.rb:12]: - [OPTIONAL] You can consider writing routes this way
```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```
as it is much shorter and it's easier to read.

**Review by gemmen29 — APPROVED** [review]
Hi @jsug9, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (jsug9) [reply config/routes.rb:12]: Dear @gemmen29 great idea! Thanks for the suggestion.

Comment (jsug9) [reply config/routes.rb:12]: Hi again @gemmen29! Could you give me a little hand? I want to implement your suggestion. It works great, however I am using author_id for my posts, which I can retrieve using GET. So, is there a way to do the same using your implementation?
Captura de Pantalla 2022-07-11 a la(s) 11 34 31 a  m

Captura de Pantalla 2022-07-11 a la(s) 11 34 46 a  m

Comment (gemmen29) [reply config/routes.rb:12]: Hi @jsug9 try this [solution](https://stackoverflow.com/questions/67371958/rails-routing-how-to-rename-the-params-hash-keys-in-a-nested-resource-block)

Comment (jsug9) [reply config/routes.rb:12]: Thanks 😁

**Pull request merged**